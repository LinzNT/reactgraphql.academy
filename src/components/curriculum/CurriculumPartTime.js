import React from 'react'
import { LinkButton } from '../buttons'
import { H2Ref } from '../text'
import Link from '../navigation/Link'
import Section, { curriedToggleNavigateTo } from './CurriculumSection'
import { Col, Row } from '../layout/Grid'
import Ul, { Li } from '../layout/Ul'
import ES6Session from './sessions/ES6Session'
import ThinkingInReactSession from './sessions/ThinkingInReactSession'
import RoutingAndDataFetchingSession from './sessions/RoutingAndDataFetchingSession'
import ReactFundamentalsRecapSession from './sessions/ReactFundamentalsRecapSession'
import FormsAndAuthSession from './sessions/FormsAndAuthSession'
import StylingInReactSession from './sessions/design/StylingInReactSession'
import IntroReduxSession from './sessions/IntroReduxSession'
import TestingIntroSession from './sessions/TestingIntroSession'
import AdvancedReduxSession from './sessions/AdvancedReduxSession'
import CurriculumCard from './CurriculumCard'
import { PART_TIME } from '../../config/data'
import selectCurriculumLayout, { LIST_TWO_COL } from './selectCurriculumLayout'
import {
  LearningObjectivesList,
  TargetAudienceList,
} from './CurriculumReactFundamentals'
import { curriculumCommonPropTypes } from './'

const PartTimeFinalProject = () => (
  <Ul>
    <Li>
      Consolidate your new React skills by adding Styled-Components and Redux to
      the application you built on session 5
    </Li>
    <Li>Discussion about architecture, features and tools</Li>
    <Li>
      Practice extreme programming and get support from the coaches and mentors
    </Li>
  </Ul>
)

const CurriculumPartTime = ({
  showTitle = true,
  isOpen,
  enableToggle,
  toggleNavigateTo = `/react/curriculum?tab=${PART_TIME}`,
  showCallToActionBottom = false,
  marketingCard = null,
  layout,
  showLinkToCurriculum = true,
  trainings,
}) => {
  const toggleNavigateToSection = curriedToggleNavigateTo(toggleNavigateTo)
  const type = PART_TIME
  const commonProps = {
    type,
    enableToggle,
    toggleNavigateTo: toggleNavigateToSection,
    isOpen,
  }
  const firstHalf = (
    <React.Fragment>
      <Section
        {...commonProps}
        title="Session 1 - Modern JavaScript"
        name="session1"
      >
        <ES6Session />
      </Section>
      <Section
        {...commonProps}
        title="Session 2 - Thinking in React"
        name="session2"
      >
        <ThinkingInReactSession />
      </Section>
      <Section
        {...commonProps}
        title="Session 3 - Routing & Data Fetching"
        name="session3"
      >
        <RoutingAndDataFetchingSession />
      </Section>
      <Section
        {...commonProps}
        title="Session 4 - Forms & Auth"
        name="session4"
      >
        <FormsAndAuthSession />
      </Section>
      <Section
        {...commonProps}
        title="Session 5 - Recap: build a React app from scratch"
        name="session5"
      >
        <ReactFundamentalsRecapSession />
      </Section>
      {marketingCard}
    </React.Fragment>
  )

  const secondHalf = (
    <React.Fragment>
      <Section
        {...commonProps}
        title="Session 6 - Styling in React"
        name="session6"
      >
        <StylingInReactSession />
      </Section>
      <Section {...commonProps} title="Session 7 - Redux" name="session7">
        <IntroReduxSession />
      </Section>
      <Section
        {...commonProps}
        title="Session 8 - Testing Fundamentals in JS"
        name="session8"
      >
        <TestingIntroSession />
      </Section>
      <Section
        {...commonProps}
        title="Session 9 - Advanced Redux"
        name="session9"
      >
        <AdvancedReduxSession />
      </Section>
      <Section
        {...commonProps}
        title="Session 10 - React Redux Real-world Project"
        name="session10"
      >
        <PartTimeFinalProject />
      </Section>
    </React.Fragment>
  )

  return (
    <React.Fragment>
      {showTitle ? (
        <Row>
          <Col lg={10} lgOffset={layout !== LIST_TWO_COL ? 1 : 0}>
            <H2Ref>
              Part-time course curriculum{' '}
              <Link to="#curriculum" name="curriculum">
                #
              </Link>
            </H2Ref>
          </Col>
        </Row>
      ) : (
        ''
      )}
      {selectCurriculumLayout({
        firstHalf,
        secondHalf,
        layout,
        type,
        trainings,
        curriculumTo: showLinkToCurriculum ? toggleNavigateTo : undefined,
      })}
      {showCallToActionBottom ? (
        <Row>
          <Col lg={10} lgOffset={1}>
            <CurriculumCard>
              <Row>
                <Col lg={5} lgOffset={1}>
                  Looking for the most complete React curriculum? Our React
                  bootcamp has it all...
                </Col>
                <Col lg={6} center>
                  <LinkButton to="/react/training/bootcamp">
                    React Bootcamp
                  </LinkButton>
                </Col>
              </Row>
            </CurriculumCard>
          </Col>
        </Row>
      ) : (
        ''
      )}
    </React.Fragment>
  )
}

CurriculumPartTime.propTypes = curriculumCommonPropTypes
CurriculumPartTime.LearningObjectivesList = LearningObjectivesList
CurriculumPartTime.TargetAudienceList = TargetAudienceList

export default CurriculumPartTime
