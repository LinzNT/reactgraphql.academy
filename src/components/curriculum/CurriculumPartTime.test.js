import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import CurriculumPartTime from './CurriculumPartTime'

describe('<CurriculumPartTime />', () => {
  it('should render component', () => {
    const wrapper = shallow(<CurriculumPartTime />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
