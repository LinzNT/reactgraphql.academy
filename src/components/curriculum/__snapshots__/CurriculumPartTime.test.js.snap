// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurriculumPartTime /> should render component 1`] = `
<Curriculum
  curriculumTo="/react/curriculum?tab=Part-time"
  firstHalf={
    Array [
      <CurriculumSection
        mt={4}
        name="session1"
        title="Session 1 - Modern JavaScript"
        toggleNavigateTo="/react/curriculum?tab=Part-time"
        type="Part-time"
      >
        <ES6Session />
      </CurriculumSection>,
      <CurriculumSection
        mt={4}
        name="session2"
        title="Session 2 - Thinking in React"
        toggleNavigateTo="/react/curriculum?tab=Part-time"
        type="Part-time"
      >
        <ThinkingInReactSession />
      </CurriculumSection>,
      <CurriculumSection
        mt={4}
        name="session3"
        title="Session 3 - Routing & Data Fetching"
        toggleNavigateTo="/react/curriculum?tab=Part-time"
        type="Part-time"
      >
        <RoutingAndDataFetchingSession />
      </CurriculumSection>,
      <CurriculumSection
        mt={4}
        name="session4"
        title="Session 4 - Forms & Auth"
        toggleNavigateTo="/react/curriculum?tab=Part-time"
        type="Part-time"
      >
        <FormsAndAuthSession />
      </CurriculumSection>,
      <CurriculumSection
        mt={4}
        name="session5"
        title="Session 5 - Recap: build a React app from scratch"
        toggleNavigateTo="/react/curriculum?tab=Part-time"
        type="Part-time"
      >
        <ReactFundamentalsRecapSession />
      </CurriculumSection>,
    ]
  }
  secondHalf={
    Array [
      <CurriculumSection
        mt={4}
        name="session6"
        title="Session 6 - Styling in React"
        toggleNavigateTo="/react/curriculum?tab=Part-time"
        type="Part-time"
      >
        <StylingInReactSession />
      </CurriculumSection>,
      <CurriculumSection
        mt={4}
        name="session7"
        title="Session 7 - Redux"
        toggleNavigateTo="/react/curriculum?tab=Part-time"
        type="Part-time"
      >
        <IntroReduxSession />
      </CurriculumSection>,
      <CurriculumSection
        mt={4}
        name="session8"
        title="Session 8 - Testing Fundamentals in JS"
        toggleNavigateTo="/react/curriculum?tab=Part-time"
        type="Part-time"
      >
        <TestingIntroSession />
      </CurriculumSection>,
      <CurriculumSection
        mt={4}
        name="session9"
        title="Session 9 - React Hooks"
        toggleNavigateTo="/react/curriculum?tab=Part-time"
        type="Part-time"
      >
        <HooksSession />
      </CurriculumSection>,
      <CurriculumSection
        mt={4}
        name="session10"
        title="Session 10 - React Redux Real-world Final Project"
        toggleNavigateTo="/react/curriculum?tab=Part-time"
        type="Part-time"
      >
        <PartTimeFinalProject />
      </CurriculumSection>,
    ]
  }
  title="Part-time course curriculum"
  type="Part-time"
/>
`;
